tests/functional/UseCase/Bucket/ShareLinkBucketTest.php

```diff
@@ -0,0 +1,359 @@
+        [$credentials2, $response2] = $this->createTestProject();
+
+        // project1 shares bucket
+        $this->sourceProjectCredentials = $credentials1;
+        $this->sourceProjectResponse = $response1;
+
+        // project2 checks the access
+        $this->targetProjectCredentials = $credentials2;
+        $this->targetProjectResponse = $response2;
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+        $this->cleanTestProject();
+    }
+
+    public function testShareAndLinkBucket(): void
+    {
+        $bucketResponse = $this->createTestBucket($this->sourceProjectCredentials);
+
+        $bucketDatabaseName = $bucketResponse->getCreateBucketObjectName();
+
+        $sourceBqClient = $this->clientManager->getBigQueryClient($this->sourceProjectCredentials);
+        $linkedBucketSchemaName = $bucketDatabaseName . '_LINKED';
+
+        $handler = new CreateTableHandler($this->clientManager);
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatabaseName;
+        $columns = new RepeatedField(GPBType::MESSAGE, CreateTableCommand\TableColumn::class);
+        $columns[] = (new CreateTableCommand\TableColumn())
+            ->setName('ID')
+            ->setType(Bigquery::TYPE_INTEGER);
+        $command = (new CreateTableCommand())
+            ->setPath($path)
+            ->setTableName('TESTTABLE_BEFORE')
+            ->setColumns($columns);
+        $handler(
+            $this->sourceProjectCredentials,
+            $command,
+            []
+        );
+        $sourceBqClient->runQuery($sourceBqClient->query(sprintf(
+            'INSERT INTO %s.`TESTTABLE_BEFORE` (`ID`) VALUES (1)',
+            BigqueryQuote::quoteSingleIdentifier($bucketDatabaseName)
+        )));
+
+        $targetProjectBqClient = $this->clientManager->getBigQueryClient($this->targetProjectCredentials);
+
+//      check that the Project2 cannot access the table yet
+        $dataset = $targetProjectBqClient->dataset($linkedBucketSchemaName);
+        $this->assertFalse($dataset->exists());
+
+        $publicPart = (array) json_decode(
+            $this->sourceProjectResponse->getProjectUserName(),
+            true,
+            512,
+            JSON_THROW_ON_ERROR
+        );
+        /** @var string $sourceProjectId */
+        $sourceProjectId = $publicPart['project_id'];
+        // share the bucket
+        $handler = new ShareBucketHandler($this->clientManager);
+        $command = (new ShareBucketCommand())
+            ->setSourceProjectId($sourceProjectId)
+            ->setSourceBucketObjectName($bucketDatabaseName)
+            ->setSourceProjectReadOnlyRoleName($this->sourceProjectResponse->getProjectReadOnlyRoleName());
+
+        $result = $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        assert($result instanceof ShareBucketResponse);
+        $listing = $result->getBucketShareRoleName();
```

Comment:
neda se predikovat jmeno toho listeningu (aka shareRole) a tim padem assertnout?

Reply:
no je to string ktory obsahuje este ciselne idcko projektu a prislo mi to zbytocne pretze to vlastne dostavame z GCP tzn ak by nieco napriklad zmenili tak by nam zacali failovat testy ale vsetko by v podstate fungovalo pretoze tu hodnotu si ulozime do db a pracujeme snou.

toto je ten listing `projects/816083521011/locations/us/dataExchanges/local_project_12_08_15_54_19_SHARE_exchanger/listings/53fe646aa8920e3374b37cbe137ddcb5in_c_Test`

Reply:
chapu, ze by to bylo bylo stale fcni, ale za me by bylo fajn zjistit, ze to google sklada nejak jinak (fix testu je pak malej PR a my na to pripadne muzeme reagovat). Ale primarne mi jde o to, ze to testuje tu odpoved.

Reply:
pridal som na to assert